core/src/main/kotlin/org/evomaster/core/search/gene/TupleGene.kt

```diff
@@ -0,0 +1,178 @@
+package org.evomaster.core.search.gene
+
+import org.evomaster.core.Lazy
+import org.evomaster.core.logging.LoggingUtil
+import org.evomaster.core.output.OutputFormat
+import org.evomaster.core.search.impact.impactinfocollection.GeneImpact
+import org.evomaster.core.search.impact.impactinfocollection.value.TupleGeneImpact
+import org.evomaster.core.search.service.AdaptiveParameterControl
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.mutator.MutationWeightControl
+import org.evomaster.core.search.service.mutator.genemutation.AdditionalGeneMutationInfo
+import org.evomaster.core.search.service.mutator.genemutation.SubsetGeneSelectionStrategy
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ *  A tuple is a fixed-size, ordered list of elements, of possible different types.
+ *  This is needed for example when representing the inputs of function calls in
+ *  GraphQL.
+ *
+ *  TODO all needed methods to make it compile
+ *
+ *  TODO double-check with Man regarding hypermutation for this gene
+ */
+class TupleGene(
+    /**
+     * The name of this gene
+     */
+    name: String,
+    /**
+     * The actual elements in the array, based on the template. Ie, usually those elements will be clones
+     * of the templated, and then mutated/randomized
+     *
+     * change `var` to `val`, need to check
+     * note that if the list of gene could be updated, its impact needs to be updated
+     */
+    val elements: List<Gene> = mutableListOf(),
+    /**
```

Comment:
why a mutable list is assigned to a non-mutable list reference?

Reply:
also considering this is a `val`, and we have a check on non-emptyness in the `init` method, i guess we should not provide a default value here

Reply:
Done.